### PR TITLE
[AIRFLOW-6062] Executor would only delete workers in its own namespace

### PR DESCRIPTION
**Pull request by dimberman**
---
Issue link: [AIRFLOW-6062](https://issues.apache.org/jira/browse/AIRFLOW-6062)
The KubernetesExecutor was only able to delete worker pods in its own namespace. This PR will match delete commands to the namespace of the pod.
 
- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by dimberman** [comment]
So I'm thinking for testing we create a second namespace in the k8s cluster and put a worker in it. @potiuk what do you think? Would that be potentially intrusive for local users?

**Comment by dimberman** [comment]
Also I honestly forgot what an absolute pleasure typing systems are :). Being able to have highlights of typing errors make handling large code bases SO much easier. #python3.7forthewin

File: airflow/executors/kubernetes_executor.py [hunk 1]
```diff
@@ -344,27 +348,33 @@ def process_error(self, event: Any) -> str:
             (raw_object['reason'], raw_object['code'], raw_object['message'])
         )
 
-    def process_status(self, pod_id: str, status: str, labels: Dict[str, str], resource_version: str) -> None:
+    def process_status(self, pod_id: str,
+                       namespace: str,
+                       status: str,
+                       labels:
```
Comment (kaxil) [inline airflow/executors/kubernetes_executor.py]:
```suggestion
                       labels: Dict[str, str]
```

File: airflow/executors/kubernetes_executor.py [hunk 2]
```diff
@@ -344,27 +348,33 @@ def process_error(self, event: Any) -> str:
...
+                       Dict[str, str],
```
Comment (kaxil) [inline airflow/executors/kubernetes_executor.py]:
```suggestion
```

**Review by kaxil — APPROVED** [review]

File: airflow/executors/kubernetes_executor.py [hunk 3]
```diff
@@ -344,27 +348,32 @@ def process_error(self, event: Any) -> str:
             (raw_object['reason'], raw_object['code'], raw_object['message'])
         )
 
-    def process_status(self, pod_id: str, status: str, labels: Dict[str, str], resource_version: str) -> None:
+    def process_status(self, pod_id: str,
+                       namespace: str,
+                       status: str,
+                       labels: Dict[str, str]
```
Comment (kaxil) [inline airflow/executors/kubernetes_executor.py]:
```suggestion
                       labels: Dict[str, str],
```

**Comment by mppetkov** [comment]
Hello, I have migrated to Airflow 1.10.10 and I am trying to use this great feature, but unfortunately when all pods from different namespace are moved to status completed, then they are not deleted. Do you have any idea whether this issues was completely fixed?

**Pull request merged**